src/trace_processor/importers/proto/winscope/protolog_parser.cc

```diff
@@ -42,17 +42,25 @@
 
 namespace perfetto::trace_processor {
 
-ProtoLogParser::ProtoLogParser(TraceProcessorContext* context)
```

Comment:
Just make the TraceProcessorContext pointer a field on the WinscopeContext? THat way you need to only pass one thing around.

Even better if you can have only the fields you need in the TraceProcessorContext as that will allow you to clearly declare your dependencies.